File: src/main/java/br/com/correios/enderecador/util/EnderecadorObservable.java
package br.com.correios.enderecador.util;

import java.util.Observable;

public class EnderecadorObservable extends Observable {
    private static EnderecadorObservable instance;

    public static EnderecadorObservable getInstance() {
        if (instance == null)
            instance = new EnderecadorObservable();
        return instance;
    }

    public void notifyObservers(Object arg) {
        setChanged();
        super.notifyObservers(arg);
    }
}

File: src/main/kotlin/br/com/correios/enderecador/util/EnderecadorObservable.kt
package br.com.correios.enderecador.util

import java.util.*

class EnderecadorObservable : Observable() {
    override fun notifyObservers(arg: Any) {
        setChanged()
        super.notifyObservers(arg)
    }

    companion object {
        var instance: EnderecadorObservable? = null
            get() {
                if (field == null) {
                    field = EnderecadorObservable()
                }
                return field
            }
            private set
    }
}
